### DISCUSSION TITLE
Unable to capture traces related to event bus operations in vertx.

### DISCUSSION THREAD

**Discussion by NikunjPatel31** [open-telemetry/opentelemetry-java-instrumentation#12817] [Q&A] [answered]
I am using the OpenTelemetry Java Agent to instrument my application, but I am unable to capture traces related to the following scenarios:

1. Internal EventBus Operations:

- My application utilizes an internal event bus where multiple events are passed between different consumers.
- Traces related to these event-driven operations are not being captured by the OpenTelemetry agent.

2. SockJS Requests:

- My application receives requests through SockJS. These requests are processed, and corresponding operations are performed.
- However, traces for these SockJS-related operations are not captured by the OpenTelemetry agent.

**Comment by viramkatudiya** [comment]
Hi ,
I am Also facing the same issue in my application as well. It would be great if we have support of vertx core in Open telemetry

**Reply by laurit** [reply]
we don't have instrumentation for vert.x event bus

**Comment by laurit** [comment, marked as answer]
> Internal EventBus Operations

it is likely that we don't have instrumentation for that, you can check from https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md whether there is instrumentation for the library that you are using

> SockJS Requests

We don't have instrumentation for websockets. I suspect that websockets are too low level, it may be possible to instrument higher level protocols implemented on top of websockets.
